### PR TITLE
Rich text improvements

### PR DESCRIPTION
**Pull request by gburgett**
* Expands the RichText struct to cover more cases
* Creates a RichTextRenderer abstract base class to render common HTML
  - When given connected services, the rich text renderer can also resolve Entry and Asset links to render `<a href` and `<img` tags.
* Enables configuring the renderer implementation in `WCC::Contentful.configure`
* Injects a connected renderer to RichText fields resolved via the ModelAPI, so that calling `#to_html` on these fields invokes the configured renderer

refs https://github.com/watermarkchurch/paper-signs/issues/3192

### CODE REVIEW

File: wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,723 @@
...
+    before do
+      require 'wcc/contentful/action_view_rich_text_renderer'
+    end
+
+    subject { WCC::Contentful::ActionViewRichTextRenderer.new(document) }
+
+    it_behaves_like 'rich text renderer'
+  end
+
+  # TODO: nokogiri implementation?
```
Comment (watermarkchurch-deployment) [inline wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb:722]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/watermarkchurch/wcc-contentful/blob/d3d02e3857247983fbdaccc9cacb876d7b0e5bba/wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb#L722"></span>TODO added in <a href="https://github.com/watermarkchurch/wcc-contentful/blob/d3d02e3857247983fbdaccc9cacb876d7b0e5bba/wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb">wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb</a> - is there a card associated with that?</td>
    </tr>
  </tbody>
</table>

File: wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,727 @@
...
```
Comment (watermarkchurch-deployment) [inline wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb:726]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/watermarkchurch/wcc-contentful/blob/b2b730d8c7365dba3689ccc0f246f1099b19c956/wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb#L726"></span>TODO added in <a href="https://github.com/watermarkchurch/wcc-contentful/blob/b2b730d8c7365dba3689ccc0f246f1099b19c956/wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb">wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb</a> - is there a card associated with that?</td>
    </tr>
  </tbody>
</table>

File: wcc-contentful-middleman/lib/wcc/contentful/middleman/extension.rb [hunk 3]
```diff
@@ -34,6 +34,8 @@ def initialize(app, options_hash = {}, &block)
         config.public_send("#{k}=", v) if config.respond_to?("#{k}=")
       end
 
+      # TODO: create MiddlemanRichTextRenderer and set as default here
```
Comment (watermarkchurch-deployment) [inline wcc-contentful-middleman/lib/wcc/contentful/middleman/extension.rb:37]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/watermarkchurch/wcc-contentful/blob/814d9aace748f0276f433814fefe4b3007b1c193/wcc-contentful-middleman/lib/wcc/contentful/middleman/extension.rb#L37"></span>TODO added in <a href="https://github.com/watermarkchurch/wcc-contentful/blob/814d9aace748f0276f433814fefe4b3007b1c193/wcc-contentful-middleman/lib/wcc/contentful/middleman/extension.rb">wcc-contentful-middleman/lib/wcc/contentful/middleman/extension.rb</a> - is there a card associated with that?</td>
    </tr>
  </tbody>
</table>

File: wcc-contentful/lib/wcc/contentful/action_view_rich_text_renderer.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+require 'action_view'
+
+# An implementation of the RichTextRenderer that uses ActionView helpers to implement content_tag and concat.
+class WCC::Contentful::ActionViewRichTextRenderer < WCC::Contentful::RichTextRenderer
+  include ActionView::Helpers::TagHelper
+  include ActionView::Helpers::TextHelper
+  include ActionView::Context
+
+  # TODO: use ActionView view context to render ERB templates for embedded entries?
```
Comment (watermarkchurch-deployment) [inline wcc-contentful/lib/wcc/contentful/action_view_rich_text_renderer.rb:11]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/watermarkchurch/wcc-contentful/blob/814d9aace748f0276f433814fefe4b3007b1c193/wcc-contentful/lib/wcc/contentful/action_view_rich_text_renderer.rb#L11"></span>TODO added in <a href="https://github.com/watermarkchurch/wcc-contentful/blob/814d9aace748f0276f433814fefe4b3007b1c193/wcc-contentful/lib/wcc/contentful/action_view_rich_text_renderer.rb">wcc-contentful/lib/wcc/contentful/action_view_rich_text_renderer.rb</a> - is there a card associated with that?</td>
    </tr>
  </tbody>
</table>

File: wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb (hunk 2 above)
Comment (watermarkchurch-deployment) [inline wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb:726]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/watermarkchurch/wcc-contentful/blob/814d9aace748f0276f433814fefe4b3007b1c193/wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb#L726"></span>TODO added in <a href="https://github.com/watermarkchurch/wcc-contentful/blob/814d9aace748f0276f433814fefe4b3007b1c193/wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb">wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb</a> - is there a card associated with that?</td>
    </tr>
  </tbody>
</table>

File: docs/1.5/wcc-contentful-middleman/WCC/Contentful/Middleman/Extension.html [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+      <span class='id identifier rubyid_config'>config</span><span class='period'>.</span><span class='id identifier rubyid_public_send'>public_send</span><span class='lparen'>(</span><span class='tstring'><span class='tstring_beg'>&quot;</span><span class='embexpr_beg'>#{</span><span class='id identifier rubyid_k'>k</span><span class='embexpr_end'>}</span><span class='tstring_content'>=</span><span class='tstring_end'>&quot;</span></span><span class='comma'>,</span> <span class='id identifier rubyid_v'>v</span><span class='rparen'>)</span> <span class='kw'>if</span> <span class='id identifier rubyid_config'>config</span><span class='period'>.</span><span class='id identifier rubyid_respond_to?'>respond_to?</span><span class='lparen'>(</span><span class='tstring'><span class='tstring_beg'>&quot;</span><span class='embexpr_beg'>#{</span><span class='id identifier rubyid_k'>k</span><span class='embexpr_end'>}</span><span class='tstring_content'>=</span><span class='tstring_end'>&quot;</span></span><span class='rparen'>)</span>
+    <span class='kw'>end</span>
+
+    <span class='comment'># TODO: create MiddlemanRichTextRenderer and set as default here
```
Comment (watermarkchurch-deployment) [inline docs/1.5/wcc-contentful-middleman/WCC/Contentful/Middleman/Extension.html:262]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/watermarkchurch/wcc-contentful/blob/4b7ad73977bb23e3609efacfef4a6001e768e62a/docs/1.5/wcc-contentful-middleman/WCC/Contentful/Middleman/Extension.html#L262"></span>TODO added in <a href="https://github.com/watermarkchurch/wcc-contentful/blob/4b7ad73977bb23e3609efacfef4a6001e768e62a/docs/1.5/wcc-contentful-middleman/WCC/Contentful/Middleman/Extension.html">docs/1.5/wcc-contentful-middleman/WCC/Contentful/Middleman/Extension.html</a> - is there a card associated with that?</td>
    </tr>
  </tbody>
</table>

File: docs/1.5/wcc-contentful/WCC/Contentful/Store/InstrumentationWrapper.html [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+41
+42
+43
+44</pre>
+    </td>
+    <td>
+      <pre class="code"><span class="info file"># File 'lib/wcc/contentful/store/instrumentation.rb', line 40</span>
+
+<span class='kw'>def</span> <span class='id identifier rubyid_find_all'>find_all</span><span class='lparen'>(</span><span class='op'>**</span><span class='id identifier rubyid_params'>params</span><span class='rparen'>)</span>
+  <span class='comment'># end happens when query is executed - todo.
```
Comment (watermarkchurch-deployment) [inline docs/1.5/wcc-contentful/WCC/Contentful/Store/InstrumentationWrapper.html:284]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/watermarkchurch/wcc-contentful/blob/4b7ad73977bb23e3609efacfef4a6001e768e62a/docs/1.5/wcc-contentful/WCC/Contentful/Store/InstrumentationWrapper.html#L284"></span>TODO added in <a href="https://github.com/watermarkchurch/wcc-contentful/blob/4b7ad73977bb23e3609efacfef4a6001e768e62a/docs/1.5/wcc-contentful/WCC/Contentful/Store/InstrumentationWrapper.html">docs/1.5/wcc-contentful/WCC/Contentful/Store/InstrumentationWrapper.html</a> - is there a card associated with that?</td>
    </tr>
  </tbody>
</table>

File: wcc-contentful-middleman/lib/wcc/contentful/middleman/extension.rb (hunk 3 above)
Comment (watermarkchurch-deployment) [inline wcc-contentful-middleman/lib/wcc/contentful/middleman/extension.rb:37]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/watermarkchurch/wcc-contentful/blob/4b7ad73977bb23e3609efacfef4a6001e768e62a/wcc-contentful-middleman/lib/wcc/contentful/middleman/extension.rb#L37"></span>TODO added in <a href="https://github.com/watermarkchurch/wcc-contentful/blob/4b7ad73977bb23e3609efacfef4a6001e768e62a/wcc-contentful-middleman/lib/wcc/contentful/middleman/extension.rb">wcc-contentful-middleman/lib/wcc/contentful/middleman/extension.rb</a> - is there a card associated with that?</td>
    </tr>
  </tbody>
</table>

File: wcc-contentful/lib/wcc/contentful/action_view_rich_text_renderer.rb (hunk 4 above)
Comment (watermarkchurch-deployment) [inline wcc-contentful/lib/wcc/contentful/action_view_rich_text_renderer.rb:11]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/watermarkchurch/wcc-contentful/blob/4b7ad73977bb23e3609efacfef4a6001e768e62a/wcc-contentful/lib/wcc/contentful/action_view_rich_text_renderer.rb#L11"></span>TODO added in <a href="https://github.com/watermarkchurch/wcc-contentful/blob/4b7ad73977bb23e3609efacfef4a6001e768e62a/wcc-contentful/lib/wcc/contentful/action_view_rich_text_renderer.rb">wcc-contentful/lib/wcc/contentful/action_view_rich_text_renderer.rb</a> - is there a card associated with that?</td>
    </tr>
  </tbody>
</table>

File: wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb (hunk 2 above)
Comment (watermarkchurch-deployment) [inline wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb:726]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/watermarkchurch/wcc-contentful/blob/4b7ad73977bb23e3609efacfef4a6001e768e62a/wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb#L726"></span>TODO added in <a href="https://github.com/watermarkchurch/wcc-contentful/blob/4b7ad73977bb23e3609efacfef4a6001e768e62a/wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb">wcc-contentful/spec/wcc/contentful/rich_text_renderer_spec.rb</a> - is there a card associated with that?</td>
    </tr>
  </tbody>
</table>

**Review by rileyjhardy — APPROVED** [review]
Looks great, thanks for the lesson!!

**Review by rileyjhardy — APPROVED** [review]
Looks great, thanks for the lesson!!

**Pull request merged**